tests/arch/arm/arm_interrupt/src/arm_interrupt.c

```diff
@@ -95,9 +95,15 @@ void test_arm_interrupt(void)
 
 			if (NVIC_GetPendingIRQ(i)) {
 				/* If the NVIC line is pending, it is
-				 * guaranteed that it is implemented.
+				 * guaranteed that it is implemented; clear the
```

Comment:
Can you elaborate, why this modification is required? (in this and in similar test suites)

Reply:
A NVIC IRQ line can be used as a level-detect non-interrupt signal. IRQ 7 (top-most IRQ on this design) is implemented as such.

Previously the tests simply verified that an unused IRQ line could be set to pending, but not that it could also be cleared. This causes problems when the given IRQ line is used as a level-detect non-interrupt signal, since the input value is latched (and since it may already have been pending when we started the routine to locate an unused IRQ line, depending on the input level supplied to that line).

The modification extends the detection logic to ensure that an unused IRQ line can both be set pending and cleared again. This makes the tests pass for this design as the detection logic will now skip IRQ 7 (the level-detect signal) and select IRQ 6 and downwards.

Reply:
Thanks for the explanation. I think you need to add some of this explanation in the logic, as a comment, to make it easy for the next reader. Sthg like: "if the NVIC line can be successfully un-pended, it is guaranteed that it can be used for software interrupt triggering" or similar...

Reply:
@ioannisg I have updated the comments in the test cases with your suggestion. Thanks.